electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,401 @@
+
+        self.device = bitbox02.BitBox02(
+            transport = u2fhid.U2FHid(hid_device),
+            device_info=bitboxes,
+            show_pairing_callback=show_pairing,
+            attestation_check_callback=attestation_check,
+        )
+
+    def bip32_list_from_keypath_string(self, bip32_path):
+        derivations = bip32_path.split('/')
+        is_mainnet = True
+        print(bip32_path)
+        if len(derivations) > 2:
+            if derivations[2] == '1\'':
+                is_mainnet = False
+        keypath = []
+        for level in derivations:
+            if level == "m":
+                continue
+            elif level[-1] == '\'':
+                keypath.append(int(level[:-1]) + HARDENED)
+            else:
+                keypath.append(int(level))
+        if len(keypath) > 1:
+            if keypath[1] == 1 + HARDENED:
+                is_mainnet = False
+
+        return keypath, is_mainnet
+
+    def get_xpub(self, bip32_path, xtype):
+        print(xtype)
+        xpub_keypath, is_mainnet = self.bip32_list_from_keypath_string(bip32_path)
+        if xtype == "p2wpkh":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                out_type = bitbox02.btc.BTCPubRequest.ZPUB
+            else:
+                coin_network = bitbox02.btc.TBTC
+                out_type = bitbox02.btc.BTCPubRequest.VPUB
+        elif xtype == "p2wpkh-p2sh":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                out_type = bitbox02.btc.BTCPubRequest.YPUB
+            else:
+                coin_network = bitbox02.btc.TBTC
+                out_type = bitbox02.btc.BTCPubRequest.UPUB
+        elif xtype == "standard":
+            coin_network = bitbox02.btc.TBTC
+            out_type = bitbox02.btc.BTCPubRequest.TPUB
+        else:
+            raise Exception("invalid keypath xtype")
+
+        xpub = self.device.btc_pub(
+                keypath = xpub_keypath,
+                output_type = out_type,
+                coin = coin_network,
+                display = False, # do not display xpub
+            )
+        print(xpub)
+        return xpub
+
+    def is_pairable(self):
+        # the bitbox should already be setup
+        if not self.device.device_info()["initialized"]:
+            return False
+        return True
+
+    def show_address(self, bip32_path, address_type):
+        print("getting address on: ", address_type, bip32_path)
+        address_keypath, is_mainnet = self.bip32_list_from_keypath_string(bip32_path)
+        if address_type == "p2wpkh" or address_type == "p2pkh" or address_type == "standard":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                script = bitbox02.btc.SCRIPT_P2WPKH
+            else:
+                coin_network = bitbox02.btc.TBTC
+                script = bitbox02.btc.SCRIPT_P2WPKH
+        elif address_type == "p2wpkh-p2sh":
+            if is_mainnet:
```

Comment:
can be made a 2-liner instead of 6-liner, as script is the same in both cases, and the other can can be done inline: `coin_network ? ... if is_mainnet else ...`